cli/lib/kontena/cli/config.rb

```diff
@@ -79,9 +79,41 @@ def extract_token!(hash={})
         )
       end
 
-      # Load configuration from default location ($HOME/.kontena_client.json)
+      def move_config
+        if File.exist?(old_default_config_filename)
+          ensure_config_path
+          require 'fileutils'
+          FileUtils.mv(old_default_config_filename, default_config_filename)
+          at_exit do
+            # Need to do at exit, otherwise it will pop up in the first place the config is accessed,
+            # often after the first lines of command output have already been displayed
+            warn(
+              Kontena.pastel.yellow('Note: ') +
+              "Configuration file " + Kontena.pastel.cyan(old_default_config_filename) +
+              " was moved to the new default location " + Kontena.pastel.cyan(default_config_filename)
+            )
+          end
+          true
+        else
+          false
+        end
+      rescue =>  ex
+        raise ex, "[ERROR] Could not move config file from #{old_default_config_filename} to new default location #{default_config_filename} : #{ex.message}"
+      end
+
+      # Load configuration from default location ($HOME/.kontena/config.json)
       def load_settings_from_config_file
-        settings = config_file_available? ? parse_config_file : default_settings
+        if config_file_available?
+          settings = parse_config_file
+        elsif ENV['KONTENA_CONFIG']
+          raise "Could not read configuration from #{ENV['KONTENA_CONFIG']}"
+        else
+          if move_config
```

Comment:
This logic is a bit convoluted to read, what with the implicit `config_file_available?  <=> ENV['KONTENA_CONFIG']` thing, and the oddly named `if move_config`